cruise-control-metrics-reporter/src/main/java/com/linkedin/kafka/cruisecontrol/metricsreporter/CruiseControlMetricsReporter.java

```diff
@@ -112,11 +119,46 @@ public void configure(Map<String, ?> configs) {
 
     _cruiseControlMetricsTopic = reporterConfig.getString(CruiseControlMetricsReporterConfig.CRUISE_CONTROL_METRICS_TOPIC_CONFIG);
     _reportingIntervalMs = reporterConfig.getLong(CruiseControlMetricsReporterConfig.CRUISE_CONTROL_METRICS_REPORTING_INTERVAL_MS_CONFIG);
+
+    if(reporterConfig.getBoolean(CruiseControlMetricsReporterConfig.CRUISE_CONTROL_METRICS_TOPIC_AUTO_CREATE_CONFIG)) {
+      _newTopic = createNewTopicFromReporterConfig(reporterConfig);
+      final Properties adminProperties = new Properties();
+      adminProperties.put(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, producerProps.get(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG));
+      adminProperties.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, producerProps.get(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG));
+      _adminClient = KafkaAdminClient.create(adminProperties);
+    }
+
+  }
+
+  private static NewTopic createNewTopicFromReporterConfig(CruiseControlMetricsReporterConfig reporterConfig) {
+    String cruiseControlMetricsTopic = reporterConfig.getString(CruiseControlMetricsReporterConfig.CRUISE_CONTROL_METRICS_TOPIC_CONFIG);
+    Integer cruiseControlMetricsTopicNumPartition = reporterConfig.getInt(CruiseControlMetricsReporterConfig.CRUISE_CONTROL_METRICS_TOPIC_NUM_PARTITIONS_CONFIG);
+    Short cruiseControlMetricsTopicReplicaFactor = reporterConfig.getShort(CruiseControlMetricsReporterConfig.CRUISE_CONTROL_METRICS_TOPIC_REPLICATION_FACTOR_CONFIG);
+    return new NewTopic(cruiseControlMetricsTopic, cruiseControlMetricsTopicNumPartition, cruiseControlMetricsTopicReplicaFactor);
+  }
+
+  private static void createCruiseControlMetricsTopic(AdminClient adminClient, NewTopic newTopic) {
```

Comment:
Would it possible to:
1. Remove `static` keyword,
2. Do not take any argument, and
3. User `_adminClient` and `_newTopic`?